app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1177,26 +1193,11 @@ public void onDialogDismissed(int dialogId) {
             });
 
             cardOptions.put(getString(R.string.addFromImage), () -> {
-                if (v.getId() == R.id.frontImageHolder) {
-                    mRequestedImage = Utils.CARD_IMAGE_FROM_FILE_FRONT;
-                } else if (v.getId() == R.id.backImageHolder) {
-                    mRequestedImage = Utils.CARD_IMAGE_FROM_FILE_BACK;
-                } else if (v.getId() == R.id.thumbnail) {
-                    mRequestedImage = Utils.CARD_IMAGE_FROM_FILE_ICON;
+                if (PermissionUtils.isNeedRequestStoragePermission(LoyaltyCardEditActivity.this)) {
+                    PermissionUtils.requestStoragePermission(LoyaltyCardEditActivity.this, PERMISSION_REQUEST_STORAGE);
```

Comment:
Sorry I dont get it, so what your suggestion? @TheLastProject

Reply:
@TheLastProject  but I think is different, because Permission storage is for pick photos from files but permission PERMISSION_REQUEST_CARD_IMAGE_FROM_FILE_FRONT is from camera right?

Reply:
It has `FROM_FILE` in the name, so no. And it's not about the name, it's about using a different variable depending on which view it is for. But looking more closely I see now that `mRequestedImage` is the same kind of ugly variable that `itemClickedView` is and I should probably clean that up first.

I don't think I'll still have time to clean this up before Hacktoberfest ends, so I'll just mark your change hacktoberfest-accepted as it's a helpful contribution even though I can't merge it just yet. I'll use it later to fix this :)

Reply:
hi @TheLastProject thank you so much to label as hacktoberfest-accepted, Okay got it I will wait to your commit first related this